src/Apitizer/Types/Filter.php

```diff
@@ -160,10 +160,16 @@ public function search($fields): self
     protected function validateInput($input)
     {
         if ($this->enums) {
-            if (!in_array($input, $this->enums)) {
+            if (!\is_array($input) && !in_array($input, $this->enums)) {
                 throw InvalidInputException::filterTypeError($this, $input);
             }
-            return TypeCaster::cast($input, $this->type, $this->format);
+            if (\is_array($input)) {
+                foreach ($input as $enum) {
+                    if (!\is_array($enum) && !in_array($enum, $this->enums)) {
+                        throw InvalidInputException::filterTypeError($this, $input);
+                    }
+                }
+            }
```

Comment:
Rather than putting the enum validation at the start of the rule, perhaps we can put them after the other rules? This way we don't need to repeat checks such as the `is_array` checks (see comments):

```
        if ($this->expectArray) {
            if (! \is_array($input)) {
                throw InvalidInputException::filterTypeError($this, $input);
            }

            // Here we can check that each element in the input array is contained in the enum
            // since we know at this point that the input is an array as expected.
            // foreach ($input as $value) { ... }

            return array_map(function ($value) {
                return TypeCaster::cast($value, $this->type, $this->format);
            }, $input);
        }

        if (\is_array($input)) {
            throw InvalidInputException::filterTypeError($this, $input);
        }

        // Here we can check that the single value input is within the enum

        return TypeCaster::cast($input, $this->type, $this->format);
```